### DISCUSSION TITLE
Welcome to myrobotlab Discussions!

### DISCUSSION THREAD

**Discussion by supertick** [MyRobotLab/myrobotlab#1167] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by kwatters** [comment]
Hello,  I am Kevin.  I help maintain the MRL code base.  I have built 2 InMoov robots.

**Reply by raspberrypi-devel** [reply]
wow very  impressive  Kevin.  can you please  just  answer  one  question  for me  i have  printed  everything  above  the  waist  and   have   installed  and   successfully  run   myrobotlab on   windows 10.  i can  connect  to  the   arduino 2560 and  the   main nervoboard and   applied  power.  i have  connected  the    eyes board with  servos attached   i  can  get  MRL  to  connect  to the  arduino  but  i cannot  get  the  eyes  to move and    gael and   cyberone  has  suggested  i look in forums  because  many others  have  had  this  issue.
i am  assuming    to get  the   eyes  to move  to  get  the  eyes.py code  and  insert/load it into  the  nervoboard/arduino  ?   do i load  it   right  after  i load   mrlcomm.ino?  if  not  when do i load  it?    THE  ANSWER TO THIS  WILL GET  ME ON MY  MERRY WAY.
THANX handfingerrobot

**Comment by supertick** [comment]
Ahoy ! .. I'm Greg/GroG, There's a Discord Server [here](https://discord.com/channels/887362610360643664/887362610360643667) , but GitHub had a shiny new "Discussions!" button, and I couldn't help but push it.  I'm one of the monkeys that develop MRL code.

**Reply by raspberrypi-devel** [reply]
hi  greg/grog    thanx  for  all your  help  you have   got  me on my way!

**Comment by Tekcann** [comment]
Hey, I'm a high school student named Şahin. We're trying to get this robot up and running in our school, but we're running into a snag. Every time we try to download the MyRobotLab program, we get an error. Sometimes, after we start the bat file, it just gets stuck on a single point and won't move forward. We're using the latest version of Java. Do you think that could be the issue? ANY HELP WOULD BE MUCH APPRECİATED.

![sad-cat-2](https://github.com/user-attachments/assets/6351cfa0-c357-4aa3-9e88-43146bb02d4e)

**Reply by hairygael** [reply]
Hello, could you paste the error? I cannot currently test if the repository has some dependencies problem at the moment.

**Reply by Tekcann** [reply]
I've managed to solve the first problem, but now I've hit a snag. I saw that some services were not working. I'm trying to download them, but I can't. The system says the service has been installed, but when I open it, there's nothing on the screen. What could be the problem? (I'm sorry I didn't see your message earlier.)

**Reply by hairygael** [reply]
I just tested the MRL version 1.1.1610 and it installs properly. The fact you are missing some services and they appear as being already installed, is because the installation of Myrobotlab didn't complete properly. You need to delete it completely and reinstall. If it seems to hang during the installation, it's because your PC is a bit slow, but you need to wait until MRL restarts and it will launch Chrome browser by itself. When a PC is slow, it can take more than 30 minutes to complete the installation.

**Reply by Tekcann** [reply]
I've tried many times, but I still can't use the services or write code in Python. Actually, I can't use any of the services. I tried deleting the .ivy2 file, I tried using a VPN, I tried installing it on different internet networks and different computers, but I encounter the same issues with all of them. The services are installed incorrectly, and afterwards, none of the services start.

**Comment by hairygael** [comment]
Can you check if the .javacpp is present in the directory
C:/Users/{YourUserName}
If it is missing download this one, unzip it and paste it in the directory
C:/Users/{YourUserName
https://drive.google.com/file/d/11-uXVVBwerO6xjlDWDo6T807UIvW5zKV/view?usp=drive_link

Gael Langevin
Creator of InMoov
InMoov Robot <http://www.inmoov.fr>
@inmoov <http://twitter.com/inmoov>

**Reply by Tekcann** [reply]
Thank you very much. I have a question: if I install mrl, how can I learn to write code? 


<img width="466" height="441" alt="image" src="https://github.com/user-attachments/assets/b9abfb30-993b-4533-acdd-0b56ead61099" />

**Reply by hairygael** [reply]
With MRL and InMoov, you can run your robot without the need to write code as many codes are already implemented. That being said if you wish to create specific codes, you can find many python examples for InMoov in the directory myrobotlab-1.1.XXXX/resource/InMoov2/Gestures.
Most of these gestures can be triggered by voice command via the chatBot, you can find examples for InMoov in the directory myrobotlab-1.1.XXXX/resource/ProgramAB/tr-TR/aiml/_inmoovGestures.aiml
